web3/auto/http.py

```diff
@@ -0,0 +1,16 @@
+from web3 import (
+    HTTPProvider,
+    Web3
+)
+
+
+def http():
+
+    w3 = Web3(HTTPProvider('http://localhost:8545'))
+    if w3.isConnected():
+        return w3
+    else:
+        return False
+
+
+w3 = http()
```

Comment:
I'm thinking that it may not be good to instantiate this here since it's done in `web3.auto`

Reply:
This supports the API from the issue:
```
from web3.auto.http import w3
```
Which would give you an autocreated http connection even if an IPC one exists. I suppose we could revisit whether that's necessary. I don't have a strong use case for it, it just seemed like a good idea in the abstract.

Reply:
Also, this comment should address the double-instantiation issue: https://github.com/pipermerriam/web3.py/pull/344#discussion_r143795348

Reply:
Thanks for the feedback. I will work on the recommendations.

^.^

Reply:
My two cents on the issue of whether to use from `web3.auto.http import w3` and  `web3.auto.ipc import w3`.

I don't think people will be running three different ethereum clients, only I was to test this implementation. They will likely be running a single client. I might be wrong on this but it is usually testrpc or parity or geth.

Reply:
I'm okay with this direction: either you're using a simple, single-client setup and can use `from web3.auto import w3` or you have multiple clients and you need to specify the provider explicitly. That doesn't seem like an unreasonable burden.